File: GithubUserJava/app/src/main/java/com/example/myapplication/viewmodel/MainViewModel.java
package com.example.myapplication.viewmodel;

import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

//import com.example.myapplication.data.local.datastore.SettingPreferences;
import com.example.myapplication.data.remote.retrofit.ApiConfig;
import com.example.myapplication.data.remote.response.ItemsItem;
import com.example.myapplication.data.remote.response.ResponseGithub;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainViewModel extends ViewModel {

    private final MutableLiveData<List<ItemsItem>> _user = new MutableLiveData<>();
    public final LiveData<List<ItemsItem>> user = _user;

    private final MutableLiveData<Boolean> _isLoading = new MutableLiveData<>();
    public final LiveData<Boolean> isLoading = _isLoading;

    private final MutableLiveData<String> _username = new MutableLiveData<>();
    public final LiveData<String> username = _username;

    private final MutableLiveData<String> _avatarUrl = new MutableLiveData<>();
    public final LiveData<String> avatarUrl = _avatarUrl;

//    private final SettingPreferences pref;
//
//    public MainViewModel(SettingPreferences pref) {
//        this.pref = pref;
//    }

    public void findUser(String username) {
        getUser(username);
    }

    private void getUser(String username) {
        _isLoading.setValue(true);
        Call<ResponseGithub> client = ApiConfig.getApiService().getUser(username);
        client.enqueue(new Callback<ResponseGithub>() {
            @Override
            public void onResponse(Call<ResponseGithub> call, Response<ResponseGithub> response) {
                _isLoading.setValue(false);
                if (response.isSuccessful()) {
                    if (response.isSuccessful()) {
                        _user.setValue(response.body().getItems());
                    }
                } else {
                    Log.e("MainViewModel", "onFailure: " + response.message());
                }
            }

            @Override
            public void onFailure(Call<ResponseGithub> call, Throwable t) {
                _isLoading.setValue(false);
                Log.e("MainViewModel", "onFailure: " + t.getMessage());
            }
        });
    }
//    public LiveData<Boolean> getThemeSettings() {
//        return pref.getThemeSetting().asLiveData;
//    }
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/viewmodel/MainViewModel.kt
package com.example.myapplication.viewmodel

import android.content.ContentValues.TAG
import android.util.Log.e
import androidx.lifecycle.*
import com.example.myapplication.data.local.datastore.SettingPreferences
import com.example.myapplication.data.remote.retrofit.ApiConfig
import com.example.myapplication.data.remote.response.ItemsItem
import com.example.myapplication.data.remote.response.ResponseGithub
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainViewModel(private val pref: SettingPreferences) : ViewModel() {

    private val _user = MutableLiveData<List<ItemsItem>>()
    val user : LiveData<List<ItemsItem>> = _user

    private val _isLoading = MutableLiveData<Boolean>()
    val isLoading: LiveData<Boolean> = _isLoading

    private val _username = MutableLiveData<String?>()
    val username : LiveData<String?> = _username

    private val _avatarUrl = MutableLiveData<String?>()
    val avatarUrl : LiveData<String?> = _avatarUrl

    fun findUser(username : String) {
        getUser(username)
    }

    private fun getUser(username: String) {
        _isLoading.value = true
        val client = ApiConfig.getApiService().getUser(username)
        client.enqueue(object : Callback<ResponseGithub> {
            override fun onResponse(
                call: Call<ResponseGithub>,
                response: Response<ResponseGithub>
            ) {
                _isLoading.value = false
                if (response.isSuccessful) {
                    if (response.isSuccessful) {
                        _user.value = response.body()?.items
                    }

                } else {
                    e(TAG, "onFailure: ${response.message()}")
                }
            }

            override fun onFailure(call: Call<ResponseGithub>, t: Throwable) {
                _isLoading.value = false
                e(TAG, "onFailure: ${t.message}")
            }

        })

    }
    fun getThemeSettings(): LiveData<Boolean> {
        return pref.getThemeSetting().asLiveData()
    }

}

